streams/src/test/java/org/apache/kafka/streams/integration/IQv2StoreIntegrationTest.java

```diff
@@ -1632,10 +1729,65 @@ public <V> void shouldHandleRangeQuery(
                     IllegalArgumentException.class,
                     queryResult.get(partition)::getFailureMessage
                 );
-
                 try (final KeyValueIterator<Integer, V> iterator = queryResult.get(partition).getResult()) {
                     while (iterator.hasNext()) {
-                        actualValues.add(valueExtactor.apply(iterator.next().value));
+                        actualValues.add((Integer) iterator.next().value);
+                    }
+                }
+                assertThat(queryResult.get(partition).getExecutionInfo(), is(empty()));
+            }
+            assertThat("Result:" + result, actualValues, is(expectedValues));
+            assertThat("Result:" + result, result.getPosition(), is(INPUT_POSITION));
+        }
+    }
+
+    public <V> void shouldHandleTimestampedRangeQuery(
+        final Optional<Integer> lower,
+        final Optional<Integer> upper,
+        final boolean isKeyAscending,
+        final List<Integer> expectedValues) {
```

Comment:
Similar to above. We should expect a list of `ValueAndTimestamp` to also verify the ts we get back.

Reply:
Cf my comment above -- you need to set the right timestamps for each result. But it should be deterministic because the used ts are set by the test when input data is written.

Reply:
ok